narwhals/utils.py

```diff
@@ -214,3 +214,85 @@ def maybe_set_index(df: T, column_names: str | list[str]) -> T:
             )
         )
     return df
+
+
+def maybe_convert_dtypes(  # noqa: PLR0913
+    df: T,
+    infer_objects: bool = True,  # noqa: FBT001, FBT002
```

Comment:
```suggestion
    df: T,
    *,
    infer_objects: bool = True,  # noqa: FBT001, FBT002
```
(and then remove the noqas)

Reply:
though actually, all of these _might_ get renamed in pandas, and then we'd be stuck

Reply:
Done.
I wonder why it doesn't fail locally.